### PR TITLE
[CLI-283] [CLI-285] Clean shutdown and startup

### PR DESCRIPTION
**Pull request by codemodify**
### Description
Adds support to stop known processes the PIDs of which are saved in the config file including the cli

### Checklist
- [x] Pull Request title is in the format `[PROJ-12345] Some useful description here...`
- [ ] ~~I have prefixed this PR with `🚧 WIP` if it is a Work In Progress (not ready to merge)~~
- [x] The git branch is in the format `PROJ-12345-some-useful-description-here`
- [x] I have added one or more reviewers
- [ ] ~~I have added/updated manual and automated unit/end-to-end tests for any code changes~~
- [ ] ~~These changes are substantial and require QA review/testing~~
- [ ] ~~I have updated documentation (including README, inline comments and docs.remote.it docs)~~
- [x] This PR only contains one isolated change (bug fix, feature, etc)
- [ ] ~~Translation strings added/updated for all user-visible text~~
- [x] I have manually reviewed this PR to make sure only the files I intended to change were changed and nothing else
- [x] I have notified reviewers on Slack by @ mentioning them in the `#cli`channel
- [x] I have moved the Jira ticket into `Resolved` state and made a note in the ticket with a link to this PR

### Ticket
https://remoteit.atlassian.net/browse/CLI-283
https://remoteit.atlassian.net/browse/CLI-285

### CODE REVIEW

File: control/endpoints/uknown.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func (thisRef Uknown) CanHandle(mt contracts.MessageType) bool {
 // Handle -
 func (thisRef Uknown) Handle(data []byte) []byte {
 	result := NewUknownMessage()
-	result.Error = fmt.Errorf("UNKNOWN-MESSAGE: %s", string(data))
+	result.Error = fmt.Sprintf("UNKNOWN-MESSAGE: %s", string(data))
```
Comment (danawoodman) [inline control/endpoints/uknown.go]: Don't you want `Errorf` here?

File: api/graphql-client.go [hunk 2]
```diff
@@ -161,7 +162,11 @@ func (thisRef GraphQLClient) FindDeviceByUID(uid string) (Device, error) {
 		}
 	}
 
-	return devices[0], nil
+	if len(devices) > 0 {
+		return devices[0], nil
+	}
+
+	return Device{}, errors.New("device not found")
```
Comment (danawoodman) [inline api/graphql-client.go]: Do we want to return an error here? How is this error used?

File: internal/processes/service_manager.go [hunk 3]
```diff
@@ -228,6 +235,20 @@ func (thisRef ServiceManager) Start(proc ServiceProcess) error {
 		},
 	})
 
+	// inform about the a new child process was created
+	pm := contracts.ProcessMessage{
```
Comment (danawoodman) [inline internal/processes/service_manager.go]: This code appears duplicated below. If possible move to a helper.

File: connections/localproxy/connectd.go [hunk 4]
```diff
@@ -55,6 +60,20 @@ func (thisRef *connectDLocalProxy) Connnect() {
 		})
 	}
 
+	// inform about the a new child process was created
```
Comment (danawoodman) [inline connections/localproxy/connectd.go]: This looks very similar to the `service_manager.go` code. If you're repeating this all over the place, please turn into helper.

**Comment by codemodify** [comment]
How to do about the other cases, like the multiport binary that has a PID, the CLI itself has a PID and there might be other opportunities to spawn something and track it. Those don't have a structure/config section. Right now the technique is about creating and tracking a PID regardless of where it was created from. Then at next start it will clean all the previous PIDs.

File: api/graphql-client.go (hunk 2 above)
Comment (codemodify) [reply api/graphql-client.go]: It is used by the github.com/remoteit/cli/connections/localproxy/through-remoteit-proxy.go
in `graphQLClient.FindDeviceByUID(initiator.UID)`

**Pull request merged**